build fails with protobuf 3.8.0

Alpine Linux edge has just updated their protobuf from 3.6.1 to 3.8.0. postmarketOS builds on top of Alpine edge and it has mir 1.1.2 packaged. Due to the soname change from protobuf, mir needs to be rebuilt. But it does not build against protobuf 3.8.0:

```
[ 20%] Building CXX object src/server/scene/CMakeFiles/mirscene.dir/broadcasting_session_event_sink.cpp.o
In file included from /home/pmos/build/src/mir-1.1.2/src/server/frontend/session_mediator.cpp:19:
In file included from /home/pmos/build/src/mir-1.1.2/src/server/frontend/session_mediator.h:22:
In file included from /home/pmos/build/src/mir-1.1.2/src/server/frontend/display_server.h:22:
/home/pmos/build/src/mir-1.1.2/src/include/common/mir/protobuf/display_server.h:37:27: error: no type named 'Closure' in namespace 'google::protobuf'
        google::protobuf::Closure* done) = 0;
        ~~~~~~~~~~~~~~~~~~^
/home/pmos/build/src/mir-1.1.2/src/include/common/mir/protobuf/display_server.h:41:27: error: no type named 'Closure' in namespace 'google::protobuf'
        google::protobuf::Closure* done) = 0;
        ~~~~~~~~~~~~~~~~~~^
...
```

Mir's `display_server.h` has not been changed since the last two years, so I expect the current version of mir to behave the same way.

I've been reading through the protobuf source, and it seems that `google::protobuf::Closure` was removed between versions. Unfortunately the git log of protobuf is [unreadable](https://github.com/protocolbuffers/protobuf/commit/d61aede89cf188367766b971f59cf57d7835d8e8) and I couldn't find anything useful online. Closure used to be defined in `src/google/protobuf/stubs/common.h`.

(CC: @z3ntu)

--- a/src/client/rpc/mir_basic_rpc_channel.cpp
+++ b/src/client/rpc/mir_basic_rpc_channel.cpp
@@ -26,6 +26,10 @@
 #include "mir/protobuf/protocol_version.h"
 #include "mir/log.h"
 
+#if GOOGLE_PROTOBUF_VERSION >= 3008000
+#include <google/protobuf/stubs/callback.h>
+#endif
+
 #include <sstream>
 
 namespace mclr = mir::client::rpc;

--- a/src/include/common/mir/protobuf/display_server.h
+++ b/src/include/common/mir/protobuf/display_server.h
@@ -20,6 +20,9 @@
 #define MIR_PROTOBUF_DISPLAY_SERVER_H_
 
 #include "mir_protobuf.pb.h"
+#if GOOGLE_PROTOBUF_VERSION >= 3008000
+#include <google/protobuf/stubs/callback.h>
+#endif
 
 namespace mir
 {

--- a/src/include/common/mir/protobuf/display_server_debug.h
+++ b/src/include/common/mir/protobuf/display_server_debug.h
@@ -20,6 +20,9 @@
 #define MIR_PROTOBUF_DISPLAY_SERVER_DEBUG_H_
 
 #include "mir_protobuf.pb.h"
+#if GOOGLE_PROTOBUF_VERSION >= 3008000
+#include <google/protobuf/stubs/callback.h>
+#endif
 
 namespace mir
 {

